assignment2/src/com/assignment2/SaveUserDetails.java

```diff
@@ -0,0 +1,198 @@
+			else
+				user.setName(fullName);
+				break;
+		}
+		while(true) {
+			System.out.println("Enter the age:");
+			int age = s.nextInt();
+			user.setAge(age);
+			break;
+		}
+		s.nextLine();
+		while(true) {
+			System.out.println("Enter the address:");
+			String address = s.nextLine();
+			if(address.isEmpty())
+				System.out.println("This Field is Required");
+			else
+				user.setAddress(address);
+				break;
+		}
+		while(true) {
+			System.out.println("Enter the Roll Number:");
+			int rollNo = s.nextInt();
+			user.setRollNo(rollNo);;
+			break;
+		}
+		s.nextLine();
+		System.out.println("Select Courses(A/B/C/D/E/F):");
+		while(true) {
+			String c= s.next();
+			if((user.courses).indexOf(c) == -1 && (c.equalsIgnoreCase("A") || c.equalsIgnoreCase("B") || c.equalsIgnoreCase("C") || c.equalsIgnoreCase("D") || c.equalsIgnoreCase("E") || c.equalsIgnoreCase("F") ))
+				(user.courses).add(c);
+			else System.out.println("Invalid! Enter Course Again!");
+			if((user.courses).size() == 4) break;
+		}
+		
+		return user;
+	}
+	
+	public void askUserToDisplay() {
+		Scanner s = new Scanner(System.in);
+		//User u = new User();
+		System.out.println("On the basis of which attribute you want to perform sorting!");
+		System.out.println("1. Full Name");
+		System.out.println("2. Age");
+		System.out.println("3. Address");
+		System.out.println("4. Roll No.");
+		System.out.println("5. Default");
+		//u.setChoice1(s.nextInt());
+		User.choice1 = s.nextInt();
+		System.out.println("In which order?");
+		System.out.println("1. Ascending");
+		System.out.println("2. Descending");
+		System.out.println("3. Default");
+		//u.setChoice2(s.nextInt());
+		User.choice2 = s.nextInt();
+	}
+	public void run() {
+		String fileName = "file.txt";
+		Scanner sc = new Scanner(System.in);
+		SaveUserDetails obj = new SaveUserDetails();
+		ArrayList<User> users = new ArrayList<User>();
+		boolean flag = true;
+		//Reading data from disk
+		try
+        {
+            FileInputStream fis = new FileInputStream(fileName);
+            ObjectInputStream ois = new ObjectInputStream(fis);
+ 
+            users = (ArrayList) ois.readObject();
+ 
+            ois.close();
+            fis.close();
+        }
+        catch (IOException ioe)
+        {
+            ioe.printStackTrace();
+            return;
+        }
+        catch (ClassNotFoundException c)
+        {
+            System.out.println("Class not found");
+            c.printStackTrace();
+            return;
+        }
+		while(flag == true) {
+			System.out.println("Select the operation you want to perform-:");
+			System.out.println("1. Add User");
+			System.out.println("2. Display Users");
+			System.out.println("3. Delete User");
+			System.out.println("4. Save Users to the disk");
+			System.out.println("5. Exit");
+			
+			System.out.println("Please Enter the number of your choice:");
+			int option = sc.nextInt();
+			switch(option) {
+			case 1: 
+					User user = null;
```

Comment:
Avoid using switch cases with a large number of logic in them, either wrap them in functions or structure the code such that readability doesn't take a hit.